hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -288,14 +327,22 @@ public InterceptorRegistry getInterceptorRegistry() {
     public void onDestroy() {
     }
 
-    public boolean shouldCloneOnEntryProcessing() {
-        return getIndexes().hasIndex() && OBJECT.equals(mapConfig.getInMemoryFormat());
+    public boolean shouldCloneOnEntryProcessing(int partitionId) {
+        return getIndexes(partitionId).hasIndex() && OBJECT.equals(mapConfig.getInMemoryFormat());
     }
 
     public ObjectNamespace getObjectNamespace() {
         return objectNamespace;
     }
 
+    public void addIndexDefinition(String name, Boolean ordered) {
+        indexDefinitions.put(name, ordered);
+    }
+
+    public Set<Map.Entry<String, Boolean>> getIndexDefinitionsEntrySet() {
+        return Collections.unmodifiableSet(indexDefinitions.entrySet());
```

Comment:
Hmm, this one actually won't work when the index is not global. I had to change it to:
```
 Map<String, Boolean> definitions = new HashMap<String, Boolean>();
        if (isGlobalIndexEnabled()) {
            for (Index index : globalIndexes.getIndexes()) {
                definitions.put(index.getAttributeName(), index.isOrdered());
            }
        } else {
            for (PartitionContainer container : mapServiceContext.getPartitionContainers()) {
                for (Index index : container.getIndexes(name).getIndexes()) {
                    definitions.put(index.getAttributeName(), index.isOrdered());
                }
            }
        }
        return definitions;
```